src/workflows/recipes/components/FilterObjectForm.js

```diff
@@ -114,6 +114,26 @@ export function deserializeFilterObjectToList(obj) {
   );
 }
 
+/** Return a string of a number such that it's formatted as a whole number if it is,
+ * and as a floating point number with "human readable precision".
+ * For example, if the number is `3` return `"3"`.
+ * If the number is `0.1` return `"0.1"`.
+ * Note, that that function rounds floating point errors. For example:
+ *
+ * > 100 * 0.007
+ * 0.70000000000001/0000
+ * > smartNumberFormatting(100 * 0.007)
+ * "0.7"
+ *
+ * This is useful since we store percentages in the backend as floating point numbers.
+ * The user might type in "10" to mean "10% or if they prefer more precision they can
+ * type in "0.7" to mean "0.7%" then we want too display it with the same amount
+ * of precision.
+ */
+export function smartNumberFormatting(n) {
+  return (Math.round(n * 100) / 100).toString();
```

Comment:
This utility function is nice a pure, so it seems like a good candidate for testing. Can you write some unit tests for it?

Reply:
Can do.

Reply:
Unit test added.